src/bp/strict_encoding.rs

```diff
@@ -374,3 +374,277 @@ impl StrictDecode for bip32::ExtendedPubKey {
         })
     }
 }
+
+#[cfg(test)]
+mod test {
+
+    use super::*;
+    use crate::bp::{short_id::Descriptor, BlockChecksum, TxChecksum};
+    use bitcoin::{
+        blockdata::script::Builder, hashes::hex::FromHex, secp256k1::Message, BlockHash,
+    };
+    use rand::{thread_rng, RngCore};
+    use std::{convert::TryFrom, str::FromStr};
+
+    fn encode_decode<T: StrictEncode + StrictDecode>(object: &T) -> Result<(T, usize), Error> {
+        let mut encoded_object: Vec<u8> = vec![];
+        let written = object.strict_encode(&mut encoded_object).unwrap();
+        let decoded_object = T::strict_decode(&encoded_object[..]).unwrap();
+        Ok((decoded_object, written))
+    }
+
+    #[test]
+    fn test_encoding_network() {
+        let mainnet_bytes = &[0xF9u8, 0xBEu8, 0xB4u8, 0xD9u8][..];
+        let testnet_bytes = &[0x0Bu8, 0x11u8, 0x09u8, 0x07u8][..];
+        let regtest_bytes = &[0xFAu8, 0xBFu8, 0xB5u8, 0xDAu8][..];
+        let signet_bytes = &[0x7Eu8, 0xC6u8, 0x53u8, 0xA5u8][..];
+        //let random_bytes = &[0xA1u8, 0xA2u8, 0xA3u8, 0xA4u8][..];
+
+        let mainnet = Network::strict_decode(mainnet_bytes).unwrap();
+        let testnet = Network::strict_decode(testnet_bytes).unwrap();
+        let regtest = Network::strict_decode(regtest_bytes).unwrap();
+        let signet = Network::strict_decode(signet_bytes).unwrap();
+        //let other = Network::strict_decode(random_bytes).unwrap();
+
+        let (decoded, written) = encode_decode(&mainnet).unwrap();
+        assert_eq!(written, 4);
+        assert_eq!(decoded, mainnet);
+
+        let (decoded, written) = encode_decode(&testnet).unwrap();
+        assert_eq!(written, 4);
+        assert_eq!(decoded, testnet);
+
+        let (decoded, written) = encode_decode(&regtest).unwrap();
+        assert_eq!(written, 4);
+        assert_eq!(decoded, regtest);
+
+        let (decoded, written) = encode_decode(&signet).unwrap();
+        assert_eq!(written, 4);
+        assert_eq!(decoded, signet);
+
+        // TODO: Network::Other failing the below assertion
+        //let (decoded, written) = encode_decode(&other).unwrap();
+        //assert_eq!(written, 4);
+        //assert_eq!(decoded, other);
+    }
+
+    #[test]
+    fn test_encoding_pubkey() {
+        static PK_BYTES: [u8; 33] = [
```

Comment:
I propose to add test vectors with the following options:
- key starting with 0x02
- ZERO_KEY (must be a constant somewhere underneath `secp256k1`)
- ONE_KEY (idem)

Reply:
ZERO_KEY is removed from secp256k1 library. OKE_KEY test is added.